google-cloud-pubsub/lib/google/cloud/pubsub/subscriber/stream.rb

```diff
@@ -0,0 +1,354 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/pubsub/service"
+require "google/cloud/errors"
+require "monitor"
+require "concurrent"
+
+
+module Google
+  module Cloud
+    module Pubsub
+      class Subscriber
+        # @private
+        class Stream
+          include MonitorMixin
+
+          ##
+          # @private Implementation attributes.
+          attr_reader :request_queue, :output_enum
+
+          ##
+          # Subscriber attributes.
+          attr_reader :subscriber
+
+          ##
+          # @private Create an empty Subscriber::Stream object.
+          def initialize subscriber
+            @subscriber = subscriber
+
+            @request_queue = nil
+            @output_enum = nil
+
+            @inventory = Inventory.new self, subscriber.inventory
```

Comment:
I have a question on how each Stream object's Inventory should be handled. Consider the following code:

```ruby
subscriber = subscription.listen streams: 4, inventory 1000 do |msg|
  puts msg.msg_id
end
```

Does each Stream object have an inventory of 1000, with a total inventory of 4000? Or does each Stream object have an inventory of 250, with a total inventory of 1000? The code is implementing the former, but I can see an argument to be made for the latter. Thoughts?